UserContext/AnonymousRequestMatcher.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/*
+ * This file is part of the FOSHttpCacheBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCacheBundle\UserContext;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestMatcherInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Matches anonymous requests.
+ *
+ * Uses HTTP authentication headers as well as session name matching.
+ */
+class AnonymousRequestMatcher implements RequestMatcherInterface
+{
+    /**
+     * Valid options:
+     * - session_name_prefix.
+     *
+     * @var array
+     */
+    private $options;
+
+    public function __construct(array $options = [])
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults(['session_name_prefix' => 'PHPSESSID']);
+
+        $this->options = $resolver->resolve($options);
+    }
+
+    public function matches(Request $request)
+    {
+        if ($request->server->has('AUTHORIZATION') ||
+            $request->server->has('HTTP_AUTHORIZATION') ||
+            $request->server->has('PHP_AUTH_USER')
```

Comment:
> The part that checks for custom sessions names should then be removed completely, right ?

yes, exactly. because we also Vary on those headers, if you send a cookie header with google analytics data in it, you break the cache anyways. this has to be handled on the caching proxy.

it looks like header based authentication is always put into the AUTHORIZATION header: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/ServerBag.php#L94-L98 so cookie and authorization should be enough